.github/workflows/deploy_tre.yml

```diff
@@ -50,100 +46,3 @@ jobs:
       TF_STATE_CONTAINER: ${{ secrets.TF_STATE_CONTAINER }}
       TRE_ADDRESS_SPACE: ${{ secrets.TRE_ADDRESS_SPACE }}
       TRE_ID: ${{ secrets.TRE_ID }}
-
-  prepare-not-main:
```

Comment:
Deleting this means there's no way to deploy without a pr and also that dispatch is only targeting the main branch. Is that what you want?

Reply:
Ah, good point. I forgot this workflow is also sometimes used to build a branch without a PR.

I was hoping that this change would allow us to remove this and simplify the workflow graph:

![image](https://user-images.githubusercontent.com/1824461/158362024-920619ca-5f56-42a5-b863-57303984ce34.png)

Do you have a sense for what the scenarios are where people would want to trigger a build for a non-`main` branch where creating a PR or draft PR wouldn't make sense?

Reply:
I have used it many times to have the ci test something for me prior to opening the pr. But that was for branches in this repo and not Forks (which is how we should do things from now).

Reply:
Ok, since the code I removed would only be invoked via `workflow_dispatch`, I'll add a new workflow that contains the removed steps. As you mention, this is unlikely to be used in the normal flow (as we'll be working in forks and so secrets wouldn't be passed to the workflow), but having it will allow us to push a branch to the main repo and test workflow changes there before merging to `main`.

Reply:
@tamirkamara  - I've added `deploy_tre_branch.yml` with the non-`main` build flow. Let me know if that will handle your requirements. Thanks!